include/pistache/http_header.h

```diff
@@ -654,6 +654,8 @@ namespace Pistache::Http::Header
         std::string value_;                                                         \
     };
 
+#define CUSTOM_HEADER(header_name) PISTACHE_CUSTOM_HEADER(header_name, #header_name)
```

Comment:
Pistache already defines several unprefixed macros, and fixing it is out of the scope of this patch, in my opinion.

Reply:
See my answer for the other point. We need a second macro name if we want to be backward compatible (unless there is a way I'm not seeing). So prefixing the macro seemed a good practice for the new name. Otherwise it would have to be `CUSTOM_HEADER_2` or something like that.

Reply:
Sorry, I misread the diff, you're right :)